pkg/deploy/apiserver/apiserver.go

```diff
@@ -102,7 +103,11 @@ func (c *AppsConfig) newV1RESTStorage() (map[string]rest.Storage, error) {
 	// TODO sort out who is using this and why.  it was hardcoded before the migration and I suspect that it is being used
 	// to serialize out objects into annotations.
 	externalVersionCodec := kapi.Codecs.LegacyCodec(schema.GroupVersion{Group: "", Version: "v1"})
-	deprecatedOpenshiftClient, err := osclient.New(c.GenericConfig.LoopbackClientConfig)
+	openshiftInternalAppsClient, err := appsclientinternal.NewForConfig(c.CoreAPIServerClientConfig)
```

Comment:
@deads2k this almost looks like a job for ClientBuilder :-) (but I don't really think we should use the controller.ClientBuilder here, maybe just create something similar for GenericServer ?

Reply:
> @deads2k this almost looks like a job for ClientBuilder :-) (but I don't really think we should use the controller.ClientBuilder here, maybe just create something similar for GenericServer ?

The client builder is special because it goes and gets different credentials.  I don't think this server needs to get different credentials to run